pkg/kubelet/volumes.go

```diff
@@ -134,6 +141,19 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 		if err != nil {
 			return nil, err
 		}
+		if hasFSGroup && builder.SupportsOwnershipManagement() {
+			path := builder.GetPath()
+
+			err = kl.chownRunner.Chown(builder.GetPath(), 0, int(fsGroup))
+			if err != nil {
+				glog.Errorf("Chown failed on %v: %v", path, err)
+			}
+
+			err = kl.chmodRunner.Chmod(builder.GetPath(), os.FileMode(0770|os.ModeSetgid))
```

Comment:
This sort of feels like it should be additive rather than absolute (g+rws) or maybe 0770 for dirs and 0650 for files?